nexusadspy/segment.py

```diff
@@ -0,0 +1,100 @@
+
+from __future__ import (
+    print_function, division, generators,
+    absolute_import, unicode_literals
+)
+
+from io import BytesIO
+from gzip import GzipFile
+import time
+import logging
+
+from nexusadspy.client import AppnexusClient
+
+
+class SegmentsUploader:
+
+    BATCH_UPLOAD_ANDROID_SPECIFIER = "8"
+    BATCH_UPLOAD_IOS_SPECIFIER = "3"
+
+    def __init__(self, users_list, segment_code, separators, member_id,
+                 credentials_path='.appnexus_auth.json'):
+        """
+        Initialize batch-upload API wrapper for appnexus.
+        :param users_list: list, List of dictionaries representing appnexus users. Every member should have fields
+            -uid: Appnexus user ID. AAID/IDFS in case of mobile
+            -timestamp: Timestamp when user entered the segment
+            -expiration (optional): Expiration timestamp for the user. Default 0.
+            -value (optional): Numerical value for the segment. Default 0.
+            -mobile_os (optional): OS used by the user. Considered to be desktop if absent.
```

Comment:
If OS is absent, the id is considered to be desktop ID. It's consistent with appnexus standard.